src/components/Checkbox/Checkbox.scss

```diff
@@ -1,82 +1,121 @@
 @import '../../styles/common';
+$control-size: rem(16px);
 
 .Checkbox {
   position: relative;
-}
-
-.error {
-  // The basic checked styles have more specificity than this, so need to
-  // basically duplicate that selector here, too.
-  // stylelint-disable-next-line selector-max-specificity
-  .Input:checked + .Backdrop,
-  .Input.Input-indeterminate + .Backdrop,
-  .Backdrop {
-    @include control-backdrop(error);
-  }
-
-  // We need to know if the input is focusing to change the border color.
-  // stylelint-disable-next-line selector-max-specificity
-  .Input:focus + .Backdrop,
-  .Input:active + .Backdrop {
-    @include control-backdrop(focused-error);
-  }
-
-  .Icon {
-    @include recolor-icon(color('red', 'dark'));
-  }
+  margin: var(--p-choice-margin, 0);
 }
 
 .Input {
   @include visually-hidden;
 
-  &:focus,
-  &:active,
-  &:checked,
-  &.Input-indeterminate {
-    // stylelint-disable-next-line selector-max-class
-    + .Backdrop {
-      background: color('indigo');
+  &:not(.globalTheming) {
+    &:focus,
+    &:active,
+    &:checked,
+    &.Input-indeterminate {
+      // stylelint-disable-next-line selector-max-class, selector-max-specificity
+      + .Backdrop {
+        background: color('indigo');
+        // stylelint-disable-next-line selector-max-class, selector-max-specificity, max-nesting-depth
+        &::after {
+          background: color('white');
+        }
+      }
     }
-  }
 
-  &:checked,
-  &.Input-indeterminate {
-    // stylelint-disable-next-line selector-max-class
-    ~ .Icon {
-      transform: translate(-50%, -50%) scale(1);
+    &:checked,
+    &.Input-indeterminate {
+      // stylelint-disable-next-line selector-max-class, selector-max-specificity
+      ~ .Icon {
+        transform: translate(-50%, -50%) scale(1);
+      }
+    }
+
+    &:focus,
+    &:active {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop(focused);
+      }
     }
-  }
 
-  &:focus,
-  &:active {
-    + .Backdrop {
-      @include control-backdrop(focused);
+    &:disabled {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop(disabled);
+      }
+      // stylelint-disable-next-line selector-max-class, selector-max-specificity
+      ~ .Icon {
+        @include recolor-icon(color('ink', 'lightest'));
+      }
     }
   }
 
-  &:disabled {
-    + .Backdrop {
-      @include control-backdrop(disabled);
+  &.globalTheming {
+    &:focus {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include focus-ring-animated($show: 1);
+      }
+    }
+    // stylelint-disable-next-line selector-max-specificity
+    &:active:not(:disabled),
+    &:checked,
+    &.Input-indeterminate {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop-dl(active);
```

Comment:
Instead of a new design-language-focused mixin, what about creating a second argument to the existing `control-backdrop` mixin like `new_design_language` that defaults to false? You could use that as a switch to enable the new styles.